popup/popup.js

```diff
@@ -39,16 +39,23 @@ chrome.storage.local.get(["websiteUrl"]).then(({ websiteUrl }) => {
 			document.querySelector("#session-image").src = user.image;
 			document.querySelector("#session-name").innerHTML = user.name;
 			document.querySelector("#session-email").innerHTML = user.email;
-
-			chrome.storage.local.set({
-				userId: user.id,
-				userName: user.name,
-				userImage: user.image
-			}).then(() => {
-				console.debug(`[popup] userId has been set to ${user.id}`);
-			}).catch(err => {
-				console.error(`[popup] Sync storage could not be set. userId: ${user.id}`, err);
+			chrome.cookies.get({ url: websiteUrl, name: '__Secure-next-auth.session-token' })
+			.then((cookie) => {
+				const tokenval = cookie.value;
+				chrome.storage.local.set({
```

Comment:
I don't think we need to handle this right away, but just wondering about a particular scenario:

If the user has signed in on our website and then downloaded the Chrome extension and not opened the popup, then they are still logged in, but their local storage values would not be set. 

This will probably mean that the API calls on their content scripts will fail. This might not be a great experience for the user. 

So if possible, we should move these mission-critical operations to the background script. Don't you think so?